### PR TITLE
[docs] Fix `EditingWithDatePickers` demo

### PR DESCRIPTION
**Pull request by k-rajat19**
- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui/mui-x/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).

Closes https://github.com/mui/mui-x/issues/15965

https://deploy-preview-15967--material-ui-x.netlify.app/x/react-data-grid/custom-columns/#date-pickers

### CODE REVIEW

**Review by michelengelen — APPROVED** [review]
LGTM, but I would like a second review on this.

**Comment by michelengelen** [comment]
Hey @k-rajat19 ... could you merge master to resolve the 'out-of-date' warning? Thanks!

File: docs/data/data-grid/custom-columns/EditingWithDatePickers.tsx [hunk 1]
```diff
@@ -67,21 +68,30 @@ function GridEditDateCell({
...
+    if (hasFocus) {
+      inputRef.current!.focus();
+    }
+  }, [hasFocus]);
+
   return (
     <Component
       value={value}
       autoFocus
+      enableAccessibleFieldDOMStructure={false}
```
Comment (LukasTy) [inline docs/data/data-grid/custom-columns/EditingWithDatePickers.tsx:91]: Please avoid showcasing behavior that will no longer be the default for v8. 🙏

Comment (k-rajat19) [reply docs/data/data-grid/custom-columns/EditingWithDatePickers.tsx]: Updated 👍

**Review by LukasTy — APPROVED** [review]
Thank you for your contribution! 🙏

**Pull request merged**